src/control/client/storage_query.go

```diff
@@ -27,27 +27,46 @@ import (
 	"golang.org/x/net/context"
 
 	mgmtpb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/pkg/errors"
 )
 
-// BioHealthQuery will return all BIO device health and I/O error stats for
-// given device UUID
-func (c *connList) BioHealthQuery(req *mgmtpb.BioHealthReq) ResultQueryMap {
-	results := make(ResultQueryMap)
+func bioHealthRequest(mc Control, req interface{}, ch chan ClientResult) {
+	prepareReq, ok := req.(*mgmtpb.BioHealthReq)
+	if !ok {
+		err := errors.Errorf(msgTypeAssert, &mgmtpb.BioHealthReq{}, req)
 
-	mc, err := chooseServiceLeader(c.controllers)
-	if err != nil {
-		results[""] = ClientBioResult{"", nil, err}
-		return results
+		mc.logger().Errorf(err.Error())
+		ch <- ClientResult{mc.getAddress(), nil, err}
+		return // type err
 	}
 
-	resp, err := mc.getSvcClient().BioHealthQuery(context.Background(), req)
+	resp, err := mc.getSvcClient().BioHealthQuery(context.Background(), prepareReq)
+	if err != nil {
+		ch <- ClientResult{mc.getAddress(), nil, err} // return comms error
+		return
+	}
 
-	result := ClientBioResult{mc.getAddress(), resp, err}
-	results[result.Address] = result
+	ch <- ClientResult{mc.getAddress(), resp, nil}
+}
 
-	return results
+func (c *connList) BioHealthQuery(req *mgmtpb.BioHealthReq) ResultMap {
+	// TODO: convert the result of the following call to ResultQueryMap
```

Comment:
What are the implications of returning a ResultMap instead of a ResultQueryMap? I'm not particularly familiar with either data type.

Reply:
It would be ideal to avoid changing this signature if we can help it. Especially if we would need to change it back in the future.